### PR TITLE
Enable Swift Package Manager by default on master channel

### PR DESCRIPTION
**Pull request by loic-sharma**
Changes:

1. Enables Swift Package Manager by default on the main/master channel
2. Fixes tests that fail if Swift Package Manager is enabled

Corresponding docs change: https://github.com/flutter/website/pull/10938

Addresses https://github.com/flutter/flutter/issues/151567

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

File: dev/devicelab/bin/tasks/build_ios_framework_module_test.dart [hunk 1]
```diff
@@ -939,15 +939,6 @@ Future<void> _checkStatic(String pathToLibrary) async {
   }
 }
 
-Future<String> _dylibSymbols(String pathToDylib) {
-  return eval('nm', <String>[
-    '-g',
```
Comment (loic-sharma) [inline dev/devicelab/bin/tasks/build_ios_framework_module_test.dart:944]: `-g` is an alias for `--extern-only`, description: `Display only global (external) symbols.`

File: dev/devicelab/bin/tasks/module_test_ios.dart [hunk 2]
```diff
@@ -703,13 +703,7 @@ Future<bool> _isAppAotBuild(Directory app) async {
     'App',
   );
 
-  final String symbolTable = await eval(
-    'nm',
-    <String> [
-      '-gU',
```
Comment (loic-sharma) [inline dev/devicelab/bin/tasks/module_test_ios.dart:709]: `-U`'s description: `Don't display undefined symbols.` This isn't strictly necessary, it reduces the output.

File: dev/devicelab/lib/tasks/plugin_tests.dart [hunk 3]
```diff
@@ -77,6 +77,11 @@ class PluginTest {
       final _FlutterProject app = await _FlutterProject.create(tempDir, options, buildTarget,
           name: 'plugintestapp', template: 'app', environment: appCreateEnvironment);
       try {
+        if (cocoapodsTransitiveFlutterDependency) {
+          section('Disable Swift Package Manager');
+          await app.disableSwiftPackageManager();
```
Comment (loic-sharma) [inline dev/devicelab/lib/tasks/plugin_tests.dart:82]: We update the build settings of pods that depend on the Flutter framework (including transitive dependencies). This test only makes sense for CocoaPods, so we need to ensure we use CocoaPods.

For more context, see: https://github.com/flutter/flutter/pull/125610



 If a CocoaPo
We update CocoaPod 
This

File: dev/devicelab/lib/tasks/plugin_tests.dart [hunk 4]
```diff
@@ -494,42 +518,55 @@ s.dependency 'AppAuth', '1.6.0'
...
+            target,
+            'Flutter',
+            'ephemeral',
+            'Packages',
+            'FlutterGeneratedPluginSwiftPackage',
+            'Package.swift'
+          ));
+          final bool swiftPackageManagerEnabled = generatedSwiftManifest.existsSync();
+
+          if (!swiftPackageManagerEnabled) {
```
Comment (loic-sharma) [inline dev/devicelab/lib/tasks/plugin_tests.dart:532]: I recommend reviewing this diff with whitespace disabled: https://github.com/flutter/flutter/pull/152049/files?diff=split&w=1
 
image

File: packages/flutter_tools/test/host_cross_arch.shard/ios_content_validation_test.dart [hunk 5]
```diff
@@ -141,7 +146,18 @@ void main() {
...
 
-          expect(outputPluginFrameworkBinary, exists);
+          // Plugins are built either as a static library (SwiftPM's default)
+          // or as a dynamic library (CocoaPods's default).
+          // If built as a dynamic library, the plugin will have a .framework.
+          // If built as static library, the plugin's symbols will be in the
+          // Runner binary.
+          final bool helloDynamic = outputPluginFrameworkBinary.existsSync();
+          final bool helloStatic = AppleTestUtils
+            .getExportedSymbols(outputRunnerBinary.path)
```
Comment (loic-sharma) [inline packages/flutter_tools/test/host_cross_arch.shard/ios_content_validation_test.dart:156]: `AppleTestUtils.getExportedSymbols` is `flutter_tools`'s helper for `nm`: https://github.com/flutter/flutter/blob/91813060c0525e4015c1c11bfd38e95165f19d3e/packages/flutter_tools/test/integration.shard/test_utils.dart#L114-L129

File: packages/flutter_tools/test/host_cross_arch.shard/ios_content_validation_test.dart [hunk 6]
```diff
@@ -141,7 +146,18 @@ void main() {
...
+            .any((String symbol) => symbol.contains('HelloPlugin') && symbol.contains('handle'));
+
+          expect(helloDynamic || helloStatic, isTrue);
+          expect(helloDynamic != helloStatic, isTrue);
```
Comment (loic-sharma) [inline packages/flutter_tools/test/host_cross_arch.shard/ios_content_validation_test.dart:160]: This assertion is meant to be a logical xor ("Either built dynamically or statically, but not both"). Technically, the `!=` is the only assertion that's needed here but I added the `||` assertion for readability.

**Review by jmagman — APPROVED** [review]
Thank you for figuring out the `nm` details.  LGTM!

**Comment by cbracken** [comment]
Reason for revert: iOS builds failing in post-submit

**Pull request merged**